Build issues when disabling KaHip

When building on my (pretty up to date) arch linux and disabling kahip, linking fails:
```
> cmake -DCMAKE_BUILD_TYPE=Release -DUSE_KAHIP=OFF ..
-- The C compiler identification is GNU 9.1.0
-- The CXX compiler identification is GNU 9.1.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Intel TBB
-- TBB interface version: 11008
-- Submodule update
-- Found OpenMP_C: -fopenmp (found version "4.5") 
-- Found OpenMP_CXX: -fopenmp (found version "4.5") 
-- Found OpenMP: TRUE (found version "4.5")  
-- Configuring done
-- Generating done
-- Build files have been written to: [...]/InertialFlowCutter/build

> make console
Scanning dependencies of target console
[ 28%] Building CXX object CMakeFiles/console.dir/src/console.cpp.o
[ 28%] Building CXX object CMakeFiles/console.dir/src/permutation.cpp.o
[ 42%] Building CXX object CMakeFiles/console.dir/src/fancy_input.cpp.o
[ 57%] Building CXX object CMakeFiles/console.dir/src/list_graph.cpp.o
[ 71%] Building CXX object CMakeFiles/console.dir/src/file_utility.cpp.o
[ 85%] Building CXX object CMakeFiles/console.dir/src/geo_pos.cpp.o
[100%] Linking CXX executable console
/usr/bin/ld: CMakeFiles/console.dir/src/console.cpp.o: undefined reference to symbol 'pthread_create@@GLIBC_2.2.5'
/usr/bin/ld: /usr/lib/libpthread.so.0: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[3]: *** [CMakeFiles/console.dir/build.make:162: console] Error 1
make[2]: *** [CMakeFiles/Makefile2:110: CMakeFiles/console.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:122: CMakeFiles/console.dir/rule] Error 2
make: *** [Makefile:131: console] Error 2
``` 

Enabling KaHip fixes the issue, probably because it adds the `-pthread` CXXFLAG at some point...

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,7 +4,7 @@ project (InertialFlowCutter)
 option (USE_KAHIP "Link KaHiP and at kahip related commands to console." ON)
 
 set (CMAKE_CXX_STANDARD 14)
-set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wpedantic -Werror=return-type -Wno-stringop-truncation")
+set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wpedantic -Werror=return-type -Wno-stringop-truncation -pthread")
 set(CMAKE_EXE_LINKER_FLAGS  "${CMAKE_EXE_LINKER_FLAGS} -lm")
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
@@ -41,9 +41,9 @@ if (USE_KAHIP)
     add_subdirectory (extern/KaHIP EXCLUDE_FROM_ALL)
 endif()
 
-set (SOURCE_FILES 
-    src/console.cpp 
-    src/fancy_input.cpp 
+set (SOURCE_FILES
+    src/console.cpp
+    src/fancy_input.cpp
     src/permutation.cpp
     src/list_graph.cpp
     src/file_utility.cpp

